### ISSUE TITLE
Context can hide or disable menu item, but keyboard shortcut still works

### ISSUE THREAD

**Issue by db-inf** [IgnorantGuru/spacefm#696]
I am using spacefm version 1.0.5, installed from Ubuntu's (Xenial) software center. I also use the trash-cli package to have trash functionality on the command line, and want to add that functionality to spacefm as well. The well known corbeille-spacefm does not offer the trash functionality I need for my secondary internal and my external hard disks.

So I first changed the keyboard shortcut for the context menu item "__Delete__" to the `Shift-Delete` combination, and assigned the `Delete` key as shortcut for a new menu item "__Trash__", that calls a custom command `trash-put`, part of trash-cli.

Now I do not need nor want trash functionality on a tmpfs filesystem that I use as ramdisk, mounted to _/media/ramdisk_ (when I delete files on that ramdisk, it is to free memory). Therefor I added context conditions that test wether the current directory begins with "/media/ramdisk".

That context works OK for showing the new menu item "__Trash__", but the assigned shortcut `Delete` still calls and executes the command, both when the menu item is disabled, and when the menu item is hidden.

I can get my desired behaviour by testing for the path "/media/ramdisk" in the command itself, in addition to the context rules. But I thought it wise to report this unexpected behaviour of the combination of shortcut key and context. I think it is a similar issue as #540.

Note: to check the `Delete` key really triggers my custom command, I added some lines to that custom command that record it's activity in files on "/media/ramdisk". I add that version as a plugin.tar.xz

[Trash.spacefm-plugin.tar.gz](https://github.com/IgnorantGuru/spacefm/files/1501142/Trash.spacefm-plugin.tar.gz)

**Comment by IgnorantGuru** [comment]
Thanks for your report - this should be noted in the Automatic Context section of the manual as a footnote.

This behavior is slightly dysfunctional but may not change.  When SpaceFM displays a menu, it does an extensive context check, including disk access, to determine if an item is disabled. It is too inefficient to do this check on every keypress.  It might be able to do it before running a command based on key shortcut - could be investigated, but I think there was a technical reason why it wasn't done.

So for the time being, the command should test for a valid running context, which is good procedure anyway.

**Comment by IgnorantGuru** [comment]
Latest alpha should test context and not run commands based on key shortcuts if hidden or disabled.  I placed this change in alpha branch to allow for more testing.  The context is tested just after a keypress occurs, if that keypress is associated with a custom command.  Testing appreciated - use the net installer to test the alpha branch.

Also changed in 5c6d2bf, the 'Use as opener for' context setting now applies to files and devices opened from the SpaceFM desktop, unlike prior versions.

The manual for alpha has been updated to reflect these changes [here](http://ignorantguru.github.io/spacefm/spacefm-manual-next-en.html#designmode-props-opener).
